CMakeLists.txt

```diff
@@ -109,27 +112,30 @@ function(trusted_firmware_build)
                -DTFM_TOOLCHAIN_PREFIX=${TFM_TOOLCHAIN_PREFIX}
     BUILD_ALWAYS True
     USES_TERMINAL_BUILD True
-    BUILD_BYPRODUCTS ${VENEERS_FILE} ${PSA_API_NS_PATH}
+    BUILD_BYPRODUCTS ${BUILD_BYPRODUCTS}
   )
 
-  # IPC mode source dependencies
-  add_library(tfm_api STATIC IMPORTED)
-  set_target_properties(tfm_api PROPERTIES
-    IMPORTED_LOCATION ${PSA_API_NS_PATH}
+  add_library(tfm_api
+    ${ZEPHYR_TFM_MODULE_DIR}/tf-m-tests/app/os_wrapper_cmsis_rtos_v2.c
   )
 
-  add_library(veneer_lib STATIC IMPORTED)
-  set_target_properties(veneer_lib PROPERTIES
-    IMPORTED_LOCATION ${VENEERS_FILE}
-  )
-
-  add_dependencies(tfm_api tfm)
-  file(MAKE_DIRECTORY ${TFM_BINARY_DIR}/generated/interface/include)
-  target_include_directories(tfm_api INTERFACE
+  target_include_directories(tfm_api
+    PRIVATE
+    ${ZEPHYR_TFM_MODULE_DIR}/tf-m-tests/CMSIS/RTOS2/Include
+    PUBLIC
     ${ZEPHYR_TFM_MODULE_DIR}/trusted-firmware-m/interface/include
-    ${TFM_BINARY_DIR}/generated/interface/include
+    INTERFACE
+    ${TFM_BINARY_DIR}/install/export/tfm/include
```

Comment:
you are already using: `${ZEPHYR_TFM_MODULE_DIR}/trusted-firmware-m/interface/include` above as PUBLIC, which has the same content (+more) as the `export/tfm/include`.

Only content in `export/tfm/include` you might need is the generated `sid.h` which is in:
```
${TFM_BINARY_DIR}/generated/interface/include/psa_manifest
```

Reply:
I need `${TFM_BINARY_DIR}/generated/interface/include/psa_manifest`
Would you prefer I remove `${TFM_BINARY_DIR}/generated/interface/include`?

Reply:
>  the convention is to include sid.h as #include "psa_manifest/sid.h"

Good to hear.

> Would you prefer I remove ${TFM_BINARY_DIR}/generated/interface/include

You need it, for users of tfm to be able to include `sid.h`.
But using the generated path instead of the install path is nicer as it will make it easier to remove the install step in the future.

Reply:
Sorry, I meant would you prefer I remove `${ZEPHYR_TFM_MODULE_DIR}/trusted-firmware-m/interface/include`?

Reply:
as you were adding `export/tfm/include` in earlier revision, then I think you need either `export/tfm/include` or `${ZEPHYR_TFM_MODULE_DIR}/trusted-firmware-m/interface/include`, right ?

And in that case, I prefer the current version with: `${ZEPHYR_TFM_MODULE_DIR}/trusted-firmware-m/interface/include`